learning/dashboard/API.docs/readme.md

```diff
@@ -0,0 +1,207 @@
+
+![](https://www.cloudmanagementinsider.com/wp-content/uploads/2020/06/cloud-APIs.png)
+
+
+- APIs are that we can use to interact with GitHub. They allow us to create and manage repositories, branches, issues, pull requests, and many more. For fetching publicly available information (like public repositories, user profiles, etc.), we can call the API.
+ 
+ - Additionally, today APIs are treated more like products than code.Modern APIs stick to specific  standards(typically HTTPS and RESET), which enables API to be developer-friendly, self-described, easily accessible and understood broadly.
+
+ # How to create an API?
+Due diligence and effort are required to build an API that other developers will want to work with and trust. These are the five steps required for high-quality API design:
+
+### 1. Plan the API :
+API specifications, like OpenAPI, provide the blueprint for your API design. It is better to think about different use cases in advance and ensure the API adheres to current API development standards.
+
+### 2. Build the API :
+ API designers prototype APIs using boilerplate code. Once the prototype is tested, developers can customize it to internal specifications.
+
+### 3. Test the API :  
+API testing is the same as software testing and must be done to prevent bugs and defects. API testing tools can be used to strength test the API against cyber attacks.
+
+### 4. Document the API :  
+While APIs are self-explanatory, API documentation acts as a guide to improve usability. Well-documented APIs that offer a range of functions and use cases tend to be more popular in a service-oriented architecture.
+
+### 5. Market the API : 
+Just as Amazon is an online marketplace for retail, API marketplaces exist for developers to buy and sell other APIs. Listing your API can allow you to monetize it.
+
+## Types of APIs
+
+There are four types of APIs commonly used in web services-public,partner,private and composite. The APIs "type" indicates the intended scope of use.
+
+![](https://apponward.com/wp-content/uploads/2023/02/How-do-api-work-1.png)
+
+### Private APIs :
+These are internal to an enterprise and only used for connecting systems and data within the business.
+
+### Public APIs :
+These are open to the public and may be used by anyone. There may or not be some authorization and cost associated with these types of APIs.
+
+### Partner APIs :
+These are only accessible by authorized external developers to aid business-to-business partnerships.
+
+### Composite APIs : 
+These combine two or more different APIs to address complex system requirements or behaviors.
+
+# How to use an API?
+#### The steps to implement a new API include:
+
+- Obtaining an API key. This is done by creating a verified account with the API provider.
```

Comment:
Is this step specific to some API provider?

Reply:
No Sir. These are general instructions. The user can choose an API based on their needs. For any API provider, the user creates an account with them and further obtains an API key.